src/wallet/wallet.ts

```diff
@@ -955,12 +978,8 @@ class HathorWalletServiceWallet extends EventEmitter implements IHathorWallet {
     this.firstConnection = true;
     this.removeAllListeners();
 
-    if (cleanStorage) {
-      wallet.cleanWallet({
-        endConnection: false,
-        connection: this.conn,
-      });
-    }
+    //XXX: Should we unsubscribe from each address?
```

Comment:
Sorry, I used these `XXX` comments to mark some issues for personal organization.
This was because the `src/wallet.js` was managing address handlers to the wallet-service, but the `this.conn.stop()` below already cleans up all handlers, so we don't actually need to unsubscribe from each address.